transport/http.go

```diff
@@ -113,13 +113,13 @@ func (h *httpTransport) Call(ctx context.Context, r *Request) (*Response, error)
 		return nil, err
 	}
 	defer resp.Body.Close()
-	if resp.StatusCode < 200 || resp.StatusCode >= 300 {
-		return nil, fmt.Errorf("HTTP call got non-success response code: %v", resp.StatusCode)
-	}
 
 	body, err := ioutil.ReadAll(resp.Body)
+	if resp.StatusCode < 200 || resp.StatusCode >= 300 {
+		return nil, fmt.Errorf("HTTP call got non-success response code: %v, body: %s", resp.StatusCode, body)
```

Comment:
`body` may not have been read fully at this point, but we can expose whatever we were able to read.